src/js/lib/search.js

```diff
@@ -0,0 +1,344 @@
+            }
+          }
+        }
+
+        builder.agg("filter", key, aggregation => {
+          return aggregation
+            .orFilter("bool", filter)
+            .agg(
+              "terms",
+              key === OBJECT_TYPE ? "object_type.keyword" : key,
+              { size: 10000 },
+              key
+            )
+        })
+      } else {
+        builder.agg(
+          "terms",
+          key === OBJECT_TYPE ? "object_type.keyword" : key,
+          { size: 10000 },
+          key
+        )
+      }
+    })
+  }
+  return facetClauses
+}
+
+export const buildOrQuery = (builder, searchType, textQuery, extraClauses) => {
+  const textFilter = emptyOrNil(textQuery) ? [] : [{ bool: textQuery }]
+  builder = builder.orQuery("bool", {
+    filter: {
+      bool: {
+        must: [
+          {
+            term: {
+              object_type: searchType
+            }
+          },
+          ...extraClauses,
+          // Add multimatch text query here to filter out non-matching results
+          ...textFilter
+        ]
+      }
+    },
+    // Add multimatch text query here again to score results based on match
+    ...textQuery
+  })
+  return builder
+}
+
+const addFacetClauseToArray = (facetClauses, facet, values) => {
+  if (
+    facet === OBJECT_TYPE &&
+    values.toString() === buildSearchQuery.toString()
+  ) {
+    return
+  }
+
+  const filterKey = facet === OBJECT_TYPE ? "object_type.keyword" : facet
+
+  facetClauses.push({
+    bool: {
+      should: values.map(value => ({
+        term: {
+          [filterKey]: value
+        }
+      }))
+    }
+  })
+}
+
+export const buildSuggestQuery = (text, suggestFields) => {
+  const suggest = {
+    text
+  }
+  suggestFields.forEach(
+    field =>
+      // $FlowFixMe: yes the fields are missing and I'm adding them
+      (suggest[field] = {
+        phrase: {
+          field:      `${field}`,
+          size:       5,
+          gram_size:  1,
+          confidence: 0.0001,
+          max_errors: 3,
+          collate:    {
+            query: {
+              source: {
+                match_phrase: {
+                  "{{field_name}}": "{{suggestion}}"
+                }
+              }
+            },
+            params: { field_name: `${field}` },
+            prune:  true
+          }
+        }
+      })
+  )
+  return suggest
+}
+
+export const buildDefaultSort = () => {
+  return [
+    { minimum_price: { order: "asc" } },
+    { default_search_priority: { order: "desc" } },
+    { created: { order: "desc" } }
+  ]
+}
+
+export const SEARCH_GRID_UI = "grid"
+export const SEARCH_LIST_UI = "list"
+
+export const searchResultToLearningResource = result => ({
+  id:          result.id,
+  title:       result.title,
+  image_src:   result.image_src,
+  object_type: result.object_type,
+  lists:       result.lists,
+  is_favorite: result.is_favorite,
+  offered_by:
+    "offered_by" in result && result.offered_by ? result.offered_by : [],
```

Comment:
Should always be OCW?

Reply:
I just realized it doesn't make any sense to show 'offered by OCW' in the UI...going to delete that :joy: